src/stan/io/deserializer.hpp

```diff
@@ -935,12 +946,335 @@ class deserializer {
+   * @tparam Sizes Types of dimensions of output
+   * @param lb Lower bound
+   * @param ub Upper bound
+   * @param sizes dimensions
+   * @return Unconstrained variable
+   */
+  template <typename Ret, typename L, typename U, typename... Sizes>
+  inline auto read_free_lub(const L& lb, const U& ub, Sizes... sizes) {
+    return stan::math::lub_free(this->read<Ret>(sizes...), lb, ub);
+  }
+
+  /**
+   * Read a serialized offset-multiplied variable and unconstrain it
+   *
+   * @tparam Ret Type of output
+   * @tparam O Type of offset
+   * @tparam M Type of multiplier
+   * @tparam Sizes Types of dimensions of output
+   * @param offset Offset
+   * @param multiplier Multiplier
+   * @param sizes dimensions
+   * @return Unconstrained variable
+   */
+  template <typename Ret, typename O, typename M, typename... Sizes>
+  inline auto read_free_offset_multiplier(const O& offset, const M& multiplier,
+                                          Sizes... sizes) {
+    return stan::math::offset_multiplier_free(this->read<Ret>(sizes...), offset,
+                                              multiplier);
+  }
+
+  /**
+   * Read a serialized unit vector and unconstrain it
+   *
+   * @tparam Ret Type of output
+   * @return Unconstrained vector
+   */
+  template <typename Ret, require_not_std_vector_t<Ret>* = nullptr>
+  inline auto read_free_unit_vector(size_t size) {
+    return stan::math::unit_vector_free(this->read<Ret>(size));
+  }
+
+  /**
+   * Read serialized unit vectors and unconstrain them
+   *
+   * @tparam Ret Type of output
+   * @tparam Sizes Types of dimensions of output
+   * @param vecsize Vector size
+   * @param sizes dimensions
+   * @return Unconstrained vectors
+   */
+  template <typename Ret, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  inline auto read_free_unit_vector(size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(read_free_unit_vector<value_type_t<Ret>>(sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Read a serialized simplex and unconstrain it
+   *
+   * @tparam Ret Type of output
+   * @return Unconstrained vector
+   */
+  template <typename Ret>
+  inline auto read_free_simplex(size_t size) {
+    return stan::math::simplex_free(this->read<Ret>(size));
+  }
+
+  /**
+   * Read serialized simplices and unconstrain them
+   *
+   * @tparam Ret Type of output
+   * @tparam Sizes Types of dimensions of output
+   * @param vecsize Vector size
+   * @param sizes dimensions
+   * @return Unconstrained vectors
+   */
+  template <typename Ret, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  inline auto read_free_simplex(size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(read_free_simplex<value_type_t<Ret>>(sizes...));
+    }
+    return ret;
+  }
```

Comment:
Does this need a

```cpp
  template <typename Ret, require_not_std_vector_t<Ret>* = nullptr>
  inline auto read_free_simplex(size_t size) {
```

Or nah because you can't have a std::vector simplex? I'm pretty sure now we don't need it

Reply:
I added it anyway for symmetry with everything else lol. Pretty sure we don't need it too cause the number of size arguments will define which thing gets used.